xapian-applications/omega/query.cc

```diff
@@ -2051,6 +2055,22 @@ eval(const string &fmt, const vector<string> &param)
 		}
 		break;
 	    }
+	    case CMD_random: {
+		int rand_number;
+		if (args.empty())
+		    rand_number = rand();
+		else if (args.size() == 1)
+		    rand_number = rand() % string_to_int(args[0]);
```

Comment:
`rand()` is generally not a good quality source of pseudo-random numbers - in C++ code, C++11 offers much better options:

https://en.cppreference.com/w/cpp/numeric/random

These also provide a way to get a completely uniform distribution across the desired range - assuming `rand()` produces a uniform distribution, using `%` introduces a slight skew if the desired range size doesn't exactly divide into `RAND_MAX` (because one extra value maps to the first values in the range).

Reply:
Added mt19937 RNG, and used uniform_int_distribution to avoid that skewing.